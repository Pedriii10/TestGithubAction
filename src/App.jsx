import { useState } from "react";
import ClearInputButton from "./ClearInputButton";
import ProductAddButtons from "./ProductAddButtons";

function App() {
  const [value, setValue] = useState("");
  const [productQuantity, setPQuantity] = useState(0);
  return (
    <div
      style={{
        padding: 10,
        display: "flex",
        gap: 10,
        flexDirection: "column",
        textAlign: "center",
      }}
    >
      <ClearInputButton value={value} onChange={setValue} placeholder="..." />
      <ProductAddButtons value={productQuantity} onChange={setPQuantity} />
    </div>
  );
}

export default App;
