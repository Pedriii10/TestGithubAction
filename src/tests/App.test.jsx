import { describe, it, expect } from "vitest";

import { render } from "@testing-library/react";
import App from "../App";

describe("App", () => {
  it("renders headline", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
