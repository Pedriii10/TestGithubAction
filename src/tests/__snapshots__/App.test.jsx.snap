// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders headline 1`] = `
<div>
  <div
    style="padding: 10px; display: flex; gap: 10px; flex-direction: column; text-align: center;"
  >
    <div>
      <input
        data-testid="theInput"
        placeholder="..."
        style="padding-right: 25px;"
        value=""
      />
    </div>
    <div
      style="display: flex; gap: 5px; align-self: center;"
    >
      <button
        disabled=""
      >
        -
      </button>
      0
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;
